### ISSUE TITLE
Save game not loading

### ISSUE THREAD

**Issue by Ayrlixx** [OrionFive/Hospitality#26]
I assume this is due to an update to one of my mods today/last last night as the game was running fine and re-loadable. New starts work no problem, seems to be a conflict between Fluffy Work tab and Hospitality but I could be mistaken as I do run a very heavily modded game.

Here's my modlist; https://i.gyazo.com/fe55d551600f961f6ca6dbf9a20d7afb.png (Since making this SS I've realised I have two Ac crafting's on and have disabled the old one.)

And the error log; https://i.gyazo.com/3a339902a855bf84162c25b307963f45.png

And in text format;

Exception from asynchronous event: System.ArgumentOutOfRangeException: Argument is out of range.
Parameter name: index
  at System.Collections.Generic.List`1[System.Int32].set_Item (Int32 index, Int32 value) [0x00000] in <filename unknown>:0 
  at Verse.DefMap`2[RimWorld.WorkGiverDef,System.Int32].set_Item (RimWorld.WorkGiverDef def, Int32 value) [0x00000] in <filename unknown>:0 
  at Fluffy_Tabs.PawnPrioritiesTracker.SetPriority (RimWorld.WorkGiverDef workgiver, Int32 priority, Int32 hour) [0x00000] in <filename unknown>:0 
  at Fluffy_Tabs.PawnPrioritiesTracker.SetPriority (RimWorld.WorkGiverDef workgiver, Int32 priority) [0x00000] in <filename unknown>:0 
  at Fluffy_Tabs.PawnPrioritiesTracker.SetPriority (Verse.WorkTypeDef worktype, Int32 priority) [0x00000] in <filename unknown>:0 
  at Fluffy_Tabs.Detours_WorkSettings._SetPriority (Verse.WorkTypeDef w, Int32 priority) [0x00000] in <filename unknown>:0 
  at Hospitality.Detouring.Pawn_WorkSettings.CheckForDisabledTypes (RimWorld.Pawn_WorkSettings _this, Verse.Pawn pawn) [0x00000] in <filename unknown>:0 
  at Hospitality.Detouring.Pawn_WorkSettings.ExposeData (RimWorld.Pawn_WorkSettings _this) [0x00000] in <filename unknown>:0 
  at Verse.PostLoadInitter.DoAllPostLoadInits () [0x00000] in <filename unknown>:0 
  at Verse.Game.LoadGame () [0x00000] in <filename unknown>:0 
  at Verse.SavedGameLoader.LoadGameFromSaveFile (System.String fileName) [0x00000] in <filename unknown>:0 
  at Verse.Root_Play.<Start>m__736 () [0x00000] in <filename unknown>:0 
  at Verse.LongEventHandler.RunEventFromAnotherThread (System.Action action) [0x00000] in <filename unknown>:0 
Verse.Log:Error(String)
Verse.LongEventHandler:RunEventFromAnotherThread(Action)
Verse.LongEventHandler:<UpdateCurrentAsynchronousEvent>m__733()

**Comment by OrionFive** [comment]
Hmm... Sounds right. They both mess with the same stuff. I'll have to look into it.

**Comment by Ayrlixx** [comment]
I did try a few re-orderings of the mod load order too, to see if putting either under/above would make a difference, it didn't seem to. Regardless I've made a new save now as I've added even more mods that I had on standby for next save, I'll let you know if I experience it again! (Want me to post the same log etc to Fluffy so he is aware of this too?)

**Comment by OrionFive** [comment]
Sure, it can't hurt. Sometimes it's much easier to fix on one side than the other. Thanks for the report!

**Comment by OrionFive** [comment]
Hm. I tried to reproduce this but couldn't.

**Comment by Ayrlixx** [comment]
Yeah I honestly have no idea what caused it, current save going just fine so unless it happens again etc

**Comment by Ayrlixx** [comment]
Okay so some info from Fluffy's github, seems it is a worktab issue. (If I had known to try his solution I would have but oh well, on to new saves and all that, I get to try out faction discovery too now :D )

But if you are interested in his response;

> FluffierThanThou commented 3 hours ago
A mod added a workgiver/worktype, which throws off WorkTab. This is a known bug which I'll fix shortly (hopefully).

For now, you can fix it by temporarily disabling workTab.

Remove WorkTab, reload RW.
Open the savegame, save it again.
Re-enable WorkTab, restart RW.
Open the save again, work settings will be reset to 'simple' mode, but it should work again.
Basically, you're forcing the game to save it without the work tab data. When you then add work tab again, it re-initializes it's own data. It's supposed to detect that the data is out of sync with the current game state on it's own and re-initialize automatically, but that appears to currently be broken.

**Comment by OrionFive** [comment]
Thanks for letting me know!
